### PR TITLE
add rms_norm api

### PR DESCRIPTION
**Pull request by zhengshengning**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

新增一个 rms_norm API，接口相关参数与Torch对齐。
目前 paddle.nn.functional.rms_norm 虽然参数与Torch一致，但是paddle当前api功能并不完善。
1. 不支持 weight为None；
2. normalized_shape只支持最后一个维度的归一化；
```
Args:
  input (Tensor): Input tensor of shape [rows, cols] or higher dimensions (flattened to 2D).
  normalized_shape(int|list|tuple): Input shape from an expected input of size.
      If it is a single integer, this module will normalize over the last dimension which is expected to be of that specific size.
  weight(Tensor, optional): The weight tensor of rms_norm. Default: None.
  epsilon(float, optional): The small value added to the variance to prevent division by zero. Default: 1e-05.
  name (str, optional): Name of the operator.
Returns:
  out (Tensor): Normalized tensor of same shape as input.
  invvar (Tensor): Tensor of shape [rows], the inverse standard deviation of each row.
```

### CODE REVIEW

**Review by zhangbo9674 — DISMISSED** [review]

File: python/paddle/nn/functional/norm.py [hunk 1]
```diff
@@ -461,6 +461,87 @@ def layer_norm(
         return helper.append_activation(layer_norm_out)
 
 
+def rms_norm(
```
Comment (jeff41404) [inline python/paddle/nn/functional/norm.py:464]: 注意增加装饰器，兼容业界用法

**Review by zrr1999 — APPROVED** [review]

**Comment by swgu98** [comment]
移动了 api 位置，v卡跑不了
